conans/paths/package_layouts/package_cache_layout.py

```diff
@@ -153,15 +153,31 @@ def package_remove(self, pref):
         # with self.update_metadata() as metadata:
         #    metadata.clear_package(pref.id)
 
+    def sources_remove(self):
+        src_folder = os.path.join(self._base_folder, SRC_FOLDER)
+        try:
+            rm_conandir(src_folder)  # This will remove the shortened path too if exists
+        except OSError as e:
+            raise ConanException("%s\n\nFolder: %s\n"
+                                 "Couldn't remove folder, might be busy or open\n"
+                                 "Close any app using it, and retry" % (src_folder, str(e)))
+        scm_folder = os.path.join(self._base_folder, SCM_SRC_FOLDER)
+        try:
+            rm_conandir(scm_folder)  # This will remove the shortened path too if exists
+        except OSError as e:
+            raise ConanException("%s\n\nFolder: %s\n"
+                                 "Couldn't remove folder, might be busy or open\n"
+                                 "Close any app using it, and retry" % (scm_folder, str(e)))
+
     def export_remove(self):
         export_folder = self.export()
         rmdir(export_folder)
-        export_src_folder = self.export_sources()
+        export_src_folder = os.path.join(self._base_folder, EXPORT_SRC_FOLDER)
         rm_conandir(export_src_folder)
         download_export = self.download_export()
         rmdir(download_export)
-        scm_folder = self.scm_sources()
-        rmdir(scm_folder)
+        scm_folder = os.path.join(self._base_folder, SCM_SRC_FOLDER)
+        rm_conandir(scm_folder)
```

Comment:
Do we need here the same copy/paste as above with try/catch?

```python
        try:
            rm_conandir(....)  # This will remove the shortened path too if exists
        except OSError as e:
            raise ConanException("%s\n\nFolder: %s\n"
                                 "Couldn't remove folder, might be busy or open\n"
                                 "Close any app using it, and retry" % (scm_folder, str(e)))
```

Reply:
This is a good question. There wasn't any check before for these folders, and I am trying not to change those things unless necessary. I think we might want to do it more systematically in the future, but probably that is a different effort than this PR.